docs/design/adminconsole/INSTANCE-DATA.md

```diff
@@ -0,0 +1,280 @@
+  * Responds with `404 Not Found` if the Id does not exist.
+  * As described in [Instance Management Worker](./INSTANCE-MANAGEMENT.md), this
+    action does the following work if the status is not already "Completed",
+    using a single database transaction:
+    * Insert into `dbo.OdsInstances`.
+    * If needed, insert into `dbo.OdsInstanceContexts` and `dbo.OdsInstanceDerivatives`.
+    * Insert into `dbo.ApiClients` to create credentials for the [Health Check Worker](./HEALTH-CHECK-WORKER.md).
+    * Insert into `dbo.ApiClientOdsInstances`.
+    * Update `adminconsole.Instance` to set:
+      * New credentials
+      * Status = "Completed"
+
+### POST /adminconsole/instances/jobs/start
+
+> [!NOTE]
+> This endpoint might not be included in the Admin API 2.3 for Admin Console 1.
+
+* **Purpose**: Start processing jobs from the `adminconsole.Instance` table.
+* **Description**:
+  * Select rows with any of these conditions:
+    * Status is "Pending" and `lockDateTime is null`.
+    * Status is "In Progress" and `lockDateTime` is expired (expiration timeout
+      value to be set in appsettings, for example 60 minutes). _This provides an
+      automated retry process_
+  * Lock rows in the `adminconsole.Instance` table for processing by setting a
+    `jobId` value (UUID) and setting column `lockDateTime` to "now".
+  * Changes the status to `In Progress`
+  * Responds with `200 OK`.
+* **Response Format**:
+
+  ```json
+  {
+    "jobId": "<int>",
+    "instances": [
+      {
+        "odsInstanceId": 1,
+        "tenantId": 1,
+        "name": "Instance #1 - 2024",
+        "instanceType": "enterprise",
+        "odsInstanceContexts": [
+        {
+          "contextKey": "schoolYearFromRoute",
+          "contextValue": "2024"
+        }
+        ],
+        "odsInstanceDerivatives": [
+        {
+          "derivativeType": "ReadReplica"
+        }
+        ]
+      }
+    ]
+  }
+  ```
+
+#### POST /adminconsole/instances/jobs/{id}/completed
+
+> [!NOTE]
+> This endpoint might not be included in the Admin API 2.3 for Admin Console 1.
+
+* **Purpose**: Mark a job as complete and perform transactional updates.
+* **Enhancements**:
+  * Accept a job completion payload.
+  * Add resultant data to tables `OdsInstances`, `OdsInstanceContext`, `OdsInstanceDerivatives` and update `adminconsole.Instance` status column to mark job as `Compelete` within a single transaction.
+  * Roll back on failure.
+  * Respond with `200 Ok`.
+
+### PUT /adminconsole/instances/{id}
+
+* Not supported at this time. Respond with `405 Method Not Allowed`.
+
+### DELETE /adminconsole/instances/{id}
+
+* Not supported at this time. Respond with `405 Method Not Allowed`.
+
+## Data Storage
+
+No modifications will be made in the `dbo.*` tables.
+
+### adminconsole.Instance
+
+| Column Name   | Type           | Nullable | Purpose                                                      |
```

Comment:
`BaseUrl` and `Name` would have their own columns or this data would be part of the `Document` column only?

Reply:
I would keep those in `document`. We might want to rename `Document` in the future, if we can come up with a more descriptive name.